File: LeetCode/java/2274_Maximum_Consecutive_Floors_Without_Special_Floors.java
import java.util.Arrays;

class Solution {
    public int maxConsecutive(int bottom, int top, int[] special) {
        int answer = 0;

        Arrays.sort(special);

        for (int i = 0; i < special.length - 1; i++) {
            answer = Math.max(answer, special[i + 1] - special[i] - 1);
        }

        answer = Math.max(answer, special[0] - bottom);
        answer = Math.max(answer, top - special[special.length - 1]);

        return answer;
    }
}

File: LeetCode/cpp/2274_Maximum_Consecutive_Floors_Without_Special_Floors.cpp
class Solution {
public:
	int maxConsecutive(int bottom, int top, vector<int>& special) {
		int answer = 0;

		sort(special.begin(), special.end());

		for (int i = 0; i < special.size() - 1; i++) {
			answer = max(answer, special[i + 1] - special[i] - 1);
		}

		answer = max(answer, special[0] - bottom);
		answer = max(answer, top - special[special.size() - 1]);

		return answer;
	}
};